### PR TITLE
[feat]: [PIE-8574]: Exposed getInputSet api in openapi with caching support

### PR DESCRIPTION
**Pull request by adivishy1**
## Describe your changes
Exposed the getInputSet api in the openapi with caching support. 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)

### CODE REVIEW

**Review by mankrit-singh — DISMISSED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by mohitg0795 — DISMISSED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: :shipit: complete! ✓

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -378,6 +378,11 @@ paths:
           in: query
           name: load_from_fallback_branch
           description: Flag to load the pipeline from the created non default branch
+        - schema:
+            type: string
+          in: header
+          description: Flag to enable loading the remote pipeline from git or git cache
```
Comment (rashminandasahoo) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:384]: How about "Specifies whether the remote pipeline should be loaded from Git or Git cache."?

**Review by rashminandasahoo — APPROVED** [review]

File: pipeline-service/contracts/openapi/v1/openapi.yaml (hunk 1 above)
Comment (adivishy1) [reply pipeline-service/contracts/openapi/v1/openapi.yaml]: done

**Review by mankrit-singh — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by rashminandasahoo — DISMISSED** [review]

**Review by mohitg0795 — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: 1. unresolved discussions.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by mankrit-singh — DISMISSED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: 1. unresolved discussions.

**Review by mankrit-singh — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by deepakbolangady — APPROVED** [review]
LGTM. Thanks.

**Review by rashminandasahoo — APPROVED** [review]

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: 1. unresolved discussions.

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: 1. unresolved discussions.

**Review by mohitg0795 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538#-:-NPX5a1R7gGs7vLwqirL:bnfp4nl)* status: 1. unresolved discussions.

**Review by mohitg0795 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: 1. unresolved discussions.

**Review by rashminandasahoo — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: 1. unresolved discussions.

**Review by rashminandasahoo — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44538)* status: :shipit: complete! ✓

**Pull request merged**